lib/fastlane/actions/gradle.rb

```diff
@@ -1,35 +1,59 @@
+require 'pathname'
+require 'shellwords'
+
 module Fastlane
   module Actions
     module SharedValues
-      GRADLE_APK_OUTPUT_PATH = :APK_OUTPUT_PATH
+      GRADLE_APK_OUTPUT_PATH = :GRADLE_APK_OUTPUT_PATH
       GRADLE_FLAVOR = :GRADLE_FLAVOR
+      GRADLE_BUILD_TYPE = :GRADLE_BUILD_TYPE
     end
 
     class GradleAction < Action
       def self.run(params)
         task = params[:task]
+        flavor = params[:flavor]
+        build_type = params[:build_type]
+
+        gradle_task = [task, flavor, build_type].join
+
+        project_dir = params[:project_dir]
+
+        gradle_path_param = params[:gradle_path] || "./gradlew"
 
-        gradle = Helper::GradleHelper.new(gradle_path: params[:gradle_path])
+        # Get the path to gradle, if it's an absolute path we take it as is, if it's relative we assume it's relative to the project_dir
+        gradle_path = if Pathname.new(gradle_path_param).absolute?
+                        File.expand_path(gradle_path_param)
+                      else
+                        File.expand_path(File.join(project_dir, gradle_path_param))
+                      end
 
-        result = gradle.trigger(task: task, flags: params[:flags])
+        # Ensure we ended up with a valid path to gradle
+        raise "Couldn't find gradlew at path '#{File.expand_path(gradle_path)}'".red unless File.exist?(gradle_path)
 
-        return result unless task.start_with?("assemble")
+        # Construct our flags
+        flags = []
+        flags << "-p #{project_dir.shellescape}"
+        flags << params[:properties].map { |k, v| "-P#{k}=#{v}" }.join(' ') unless params[:properties].nil?
+        flags << params[:flags] unless params[:flags].nil?
+
+        gradle = Helper::GradleHelper.new(gradle_path: gradle_path)
+
+        result = gradle.trigger(task: gradle_task, flags: flags.join(' '))
+
+        return result unless task == 'assemble'
```

Comment:
This functionality hasn't changed since the previous version (see line 16). 

Thinking about it, does it even make sense to set a `Actions.lane_context[SharedValues::GRADLE_APK_OUTPUT_PATH]` unless assembling, as there'd be no APK?